docs/collections.md

```diff
@@ -6,29 +6,110 @@ tags:
+While [pagination](/docs/pagination/) allows you do iterate over a data set to create multiple templates, a collection allows you to group content in interesting ways. By default, Eleventy will automatically create collections for documents with a `tags` key in their front matter. One document can be part of multiple collections. You can also manually create collections by using the `addCollection` method in Eleventy’s configuration file.
 
-## A Blog Example
 
-For a blog site, your individual post files may use a tag called `post`, but it can be whatever you want. In this example, `mypost.md` has a single tag `post`:
+
+## Table of contents
+
+* [Creating tag-based collections](#creating-tag-based-collections)
+  * [Option 1: Document front matter](#option-1%3A-document-front-matter)
+  * [Option 2: Data directory files](#option-2%3A-data-directory-files)
+  * [Example: A list of blog posts](#example%3A-a-list-of-blog-posts)
+  * [Example: Navigation links with an `active` class for the current page](#example%3A-navigation-links-with-an-active-class-for-the-current-page)
+* [The special `all` collection](#the-special-all-collection)
+  * [Example: A list of links to all Eleventy-generated content](#example%3A-a-list-of-links-to-all-eleventy-generated-content)
+* [Exclude content from collections](#exclude-content-from-collections)
+* [Data structure of collection items](#data-structure-of-collection-items)
+* [Sorting collections](#sorting-collections)
+  * [Sort descending](#sort-descending)
+  * [Overriding content dates](#overriding-content-dates)
+* [Advanced: Custom filtering and sorting](#advanced%3A-custom-filtering-and-sorting)
+  * [Return values](#return-values)
+  * [Collection API methods](#collection-api-methods)
+    * [getAll()](#getall())
+    * [getAllSorted()](#getallsorted())
+    * [getFilteredByTag( tagName )](#getfilteredbytag(-tagname-))
+    * [getFilteredByGlob( glob )](#getfilteredbyglob(-glob-))
+
+
+
+## Creating tag-based collections
+
+### Option 1: Document front matter
+
+You can add a document to a collection via the `tags` key in its front matter.
 
 {% codetitle "Markdown", "Syntax" %}
 
 ```
 ---
-tags: post
+tags: posts
+---
+# Incredible article
+```
+
+In a template, you can then access this collection as `collections.posts`.
+
+The snippet above uses a string value to specify a single tag, but Eleventy also accepts array values for the `tags` key. This way, you can specify multiple tags for a document.
+
+{% codetitle "Markdown", "Syntax" %}
+
+```
+---
+tags: ['animals', 'dogs']
+---
+```
+
+{% codetitle "Markdown", "Syntax" %}
+
+```
+---
+tags:
+  - animals
+  - cats
+---
+```
+
+Both snippets function identically. While one adds a document to the “animals” and “dogs” collections, the other adds it to the “animals” and “cats” collections.
+
+
+
+### Option 2: Data directory files
```

Comment:
I think we should probably add at least a note here with a link to the docs about deep data merging https://www.11ty.io/docs/config/#data-deep-merge Although it makes almost no sense for Data Deep Merge to be buried in the Config docs. It should be in the Using Data section. Hmm

Reply:
I added a short note about this option. What do you think?